File: compiler/bench/simd3.c
// expected: 9999
// Min/max reduction - GCC vectorizes reductions
#include <stdio.h>
#define SIZE 10000
long arr[SIZE];

int main() {
    for (int i = 0; i < SIZE; i++) arr[i] = SIZE - i - 1;
    long mx = 0;
    for (int i = 0; i < SIZE; i++) if (arr[i] > mx) mx = arr[i];
    printf("%ld\n", mx);
    return 0;
}

File: compiler/bench/simd3.fs
\ expected: 9999
\ Min/max reduction - GCC vectorizes reductions

create arr 80000 allot

: init ( -- )
  10000 0 do
    10000 i - 1- i 8 * arr + !
  loop ;

: find-max ( -- n )
  0 10000 0 do
    i 8 * arr + @ max
  loop ;

: main init find-max . cr ;
